src/Program/Settings.hs

```diff
@@ -106,13 +106,29 @@ getProp name = asLikeJS . js_getProp name
 foreign import javascript unsafe "$2[$1]"
     js_getProp :: JSString -> JSVal -> JSVal
 
+-- To get a prop from "properties"
+getProperty :: LikeJS s a => JSString -> JSVal -> Maybe a
+getProperty name = asLikeJS . js_getProperty name
+
+foreign import javascript unsafe "($2.hasOwnProperty('properties') && $2['properties'] &&\
+                                 \ $2['properties'].hasOwnProperty($1)) ? $2['properties'][$1] : null"
+    js_getProperty :: JSString -> JSVal -> JSVal
+
 {-# INLINE setProp #-}
 setProp :: LikeJS s a => JSString -> a -> JSVal -> JSVal
 setProp name = js_setProp name . asJSVal
 
+setPropMaybe :: LikeJS s a => JSString -> Maybe a -> JSVal -> JSVal
+setPropMaybe name val = case val of
+                          Just v -> js_setProp name (asJSVal v)
+                          Nothing -> id
+
 foreign import javascript unsafe "$3[$1] = $2; $r = $3;"
     js_setProp :: JSString -> JSVal -> JSVal -> JSVal
 
+foreign import javascript unsafe "var r = {}; Object.assign(r, $2); delete r[$1]; $r = r"
```

Comment:
Warning! Object.assign may be not supported on internet explorer https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Object/assign
Do you have a windows computer around to test it?
If it does not, we should make a polyfill manually and put it into jsbits folder